### PR TITLE
Added armeria experiment workflow

### PR DESCRIPTION
**Pull request by ribafish**
Added armeria experiment workflow. 

Note: we might need to use the larger github runner for this as well, as they run their CI on github actions using a custom runner with 8 cores.

### CODE REVIEW

File: .github/workflows/run-experiments-armeria.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  schedule:
+    # Every Sunday at 9.00am
+    - cron: "0 9 * * 0"
+
+  workflow_dispatch:
+
+env:
+  GRADLE_ENTERPRISE_URL: "https://ge.solutions-team.gradle.com"
+  GIT_REPO: "https://github.com/line/armeria"
+  TASKS: "build -Pretry=true -PfailOnPassedAfterRetry=false -PnoLint -PflakyTests=false -PbuildJdkVersion=19 -PtestJavaVersion=19"
```
Comment (runningcode) [inline .github/workflows/run-experiments-armeria.yml:13]: is this sort of a copy of this execution?
https://github.com/line/armeria/blob/main/.github/workflows/actions_build.yml#L109

can you tell me about the `-PfailOnPassedAfterRetry=false` flag?

Comment (ribafish) [reply .github/workflows/run-experiments-armeria.yml:13]: No, it's actually based on https://github.com/line/armeria/pull/5172/files/06ae32e4635de5c5fc214ab8fb46fa310f75196a, which is armeria adding the BVS experiments (not merged yet) + adding the specified jvm as you linked it. But in general, they are aware that they some flaky tests and they wanted to not fail the runs if flakiness is detected.

Comment (runningcode) [reply .github/workflows/run-experiments-armeria.yml:13]: Got it!

**Review by runningcode — APPROVED** [review]
LGTM!

**Pull request merged**